lib/Dialect/FIRRTL/Transforms/IMConstProp.cpp

```diff
@@ -505,6 +505,8 @@ void IMConstPropPass::visitConnect(ConnectOp connect) {
     if (!AnnotationSet::get(blockArg).hasDontTouch())
       for (auto userOfResultPort : resultPortToInstanceResultMapping[blockArg])
         mergeLatticeValue(userOfResultPort, srcValue);
+    // Output ports are wire-like and may have users.
+    mergeLatticeValue(connect.dest(), srcValue);
```

Comment:
Can we move this to line 501, out of all the conditions ? That should handle all the cases.

Reply:
Yes, probably. I'll leave this for an NFC cleanup after this lands.